azure/packages/test/scenario-runner/src/DocCreatorRunner.ts

```diff
@@ -83,10 +109,130 @@ export class DocCreatorRunner extends TypedEventEmitter<IRunnerEvents> implement
 		try {
 			await Promise.all(children);
 		} catch (error) {
-			throw new Error(`Not all clients closed sucesfully.\n${error}`);
+			throw new Error(`Not all clients closed successfully.\n${error}`);
+		}
+
+		return this.docIds.length > 1 ? this.docIds : this.docIds[0];
+	}
+
+	public async runSync(config: IRunConfig): Promise<string | string[] | undefined> {
+		this.status = "running";
+		const connection = this.c.connectionConfig;
+		const connType = connection.type;
+		const connEndpoint = connection.endpoint;
+		const tenantId = connection.tenantId;
+		const tenantKey = connection.key;
+		const functionUrl = connection.functionUrl;
+		const secureTokenProvider = connection.useSecureTokenProvider;
+		const schema = this.c.schema;
+		const client = this.c.client;
+		const runs: Promise<string>[] = [];
+		for (let i = 0; i < this.c.numDocs; i++) {
+			runs.push(
+				DocCreatorRunner.execRun({
+					...config,
+					childId: i,
+					connType,
+					connEndpoint,
+					tenantId,
+					tenantKey,
+					functionUrl,
+					secureTokenProvider,
+					schema,
+					client,
+				}),
+			);
+		}
+		try {
+			const ids = await Promise.all(runs);
+			this.status = "success";
+			return ids.length > 1 ? ids : ids[0];
+		} catch {
+			this.status = "error";
+			throw new Error("Client did not close successfully.");
+		}
+	}
+
+	public static async execRun(runConfig: DocCreatorRunnerRunConfig): Promise<string> {
+		let schema;
+		const logger = await getLogger(
+			{
+				runId: runConfig.runId,
+				scenarioName: runConfig.scenarioName,
+				namespace: "scenario:runner:DocCreator",
+				endpoint: runConfig.connEndpoint,
+				region: runConfig.region,
+			},
+			["scenario:runner"],
+			eventMap,
+		);
+
+		const ac =
+			runConfig.client ??
+			(await createAzureClient({
```

Comment:
I guess this is where differentiation between async and sync sits. I did not think about this problem space too much, but I would be curious how we can incorporate some of this typing/checks on our yaml configs. With wrong configs, this will all be run-time failures, unexpected outcomes or false positives.

Reply:
Yes, my main goal here was to leave the standard flow (like you have it right now) unchanged, so your scenario runner test pipelines would continue behaving the same. I think we can come back to better enforcement once we actually have a scenario setup that requires sync (not async)